File: app/src/main/java/com/VegaSolutions/lpptransit/lppapi/responseobjects/Bus.java
package com.VegaSolutions.lpptransit.lppapi.responseobjects;

import com.google.android.gms.maps.model.LatLng;
import com.google.gson.annotations.SerializedName;

public class Bus {

    @SerializedName("bus_unit_id")
    private String busUnitId;

    private String name;
    private String vin;
    private String timestamp;

    @SerializedName("coordinate_x")
    private double coordinateX;

    @SerializedName("coordinate_y")
    private double coordinateY;

    @SerializedName("coordinate_z")
    private double coordinateZ;

    @SerializedName("cardinal_direction")
    private float cardinalDirection;

    @SerializedName("ground_speed")
    private float groundSpeed;

    @SerializedName("ignition_value")
    private boolean ignitionValue;

    @SerializedName("engine_value")
    private boolean engineValue;

    private String driverId;
    private int odo;

    public String getBusUnitId() {
        return busUnitId;
    }

    public String getName() {
        return name;
    }

    public String getVin() {
        return vin;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public double getCoordinateX() {
        return coordinateX;
    }

    public double getCoordinateY() {
        return coordinateY;
    }

    public double getCoordinateZ() {
        return coordinateZ;
    }

    public float getCardinalDirection() {
        return cardinalDirection;
    }

    public float getGroundSpeed() {
        return groundSpeed;
    }

    public boolean isIgnitionValue() {
        return ignitionValue;
    }

    public boolean isEngineValue() {
        return engineValue;
    }

    public String getDriverId() {
        return driverId;
    }

    public int getOdo() {
        return odo;
    }

    public LatLng getLatLng() {
        return new LatLng(coordinateY, coordinateX);
    }
}

File: app/src/main/java/com/VegaSolutions/lpptransit/data/lppapi/data/Bus.kt
package com.VegaSolutions.lpptransit.data.lppapi.data

import com.google.android.gms.maps.model.LatLng
import com.google.gson.annotations.SerializedName

data class Bus(
    @SerializedName("bus_unit_id")
    val busUnitId: String,

    @SerializedName("name")
    val name: String,

    @SerializedName("vin")
    val vin: String,

    @SerializedName("timestamp")
    val timestamp: String,

    @SerializedName("coordinate_x")
    val coordinateX: Double,

    @SerializedName("coordinate_y")
    val coordinateY: Double,

    @SerializedName("coordinate_z")
    val coordinateZ: Double,

    @SerializedName("cardinal_direction")
    val cardinalDirection: Float,

    @SerializedName("ground_speed")
    val groundSpeed: Float,

    @SerializedName("ignition_value")
    val isIgnitionValue: Boolean,

    @SerializedName("engine_value")
    val isEngineValue: Boolean,

    @SerializedName("driver_id")
    val driverId: String,

    @SerializedName("odo")
    val odo: Int,
) {
    val latLng: LatLng
        get() = LatLng(coordinateY, coordinateX)
}
